### PR TITLE
Fix accessibility

### PR DESCRIPTION
**Pull request by henrytao-me**
This PR attempts to fix accessibility issue in onboarding flow and navigation between screen. 

## Test cases
- ✅ Fresh start app with language already selected -> it should read out 'Step 1 of 6' on the onboarding screen
- ✅ Go to Step 3 and select 'Learn more about how it works' then close the window after it opens -> it should read out the title again `Step 3 of 6`.
- ✅ Open bottom sheet -> it should read 'Diagnosed with COVID-19?' 
- ✅ Go to code enter screen and select 'Need a one-time key?'  -> it should read 'Let people know they may have been exposed' .
- ✅ Press cancel from `Need a one-time key` screen -> back to bottom sheet -> it should read 'Diagnosed with COVID-19?'.

Ref 
- https://github.com/cds-snc/covid-shield-mobile/issues/689
- https://github.com/cds-snc/covid-shield-mobile/pull/825

### CODE REVIEW

**Comment by smcmurtry** [comment]
> ✅ Fresh start app with language already selected -> it should read out 'Step 1 of 6' on the onboarding screen
✅ Open bottom sheet -> it should read 'Diagnosed with COVID-19?'
✅ Go to code enter screen and select 'Need a one-time key?' -> it should read 'Let people know they may have been exposed' .
✅ Press cancel from Need a one-time key screen -> back to bottom sheet -> it should read 'Diagnosed with COVID-19?'.

These were all already fixed in master as of last night, by increasing the autofocus timeout from 200ms to 500ms.

> ✅ Go to Step 3 and select 'Learn more about how it works' then close the window after it opens -> it should read out the title again Step 3 of 6.
This was broken in master (used to read "Step 4 of 6": https://youtu.be/_-sbDVkoqj0) and is fixed in this PR based on my testing.

File: src/shared/useAccessibilityAutoFocus.ts [hunk 1]
```diff
@@ -24,34 +24,42 @@ export const focusOnElement = (elementRef: any) => {
...
-    callable();
-    return cancelable;
-  }, [callable, cancelable, autoFocusRef, isActive]);
+    // Call focus as soon as all considition is met
+    focusOnElement(autoFocusRef);
 
-  return [autoFocusRef, setAutoFocusRef];
-};
+    // Attempt to call it again just in case AccessibilityInfo.setAccessibilityFocus is delayed
+    const timeoutId = setTimeout(() => {
```
Comment (smcmurtry) [inline src/shared/useAccessibilityAutoFocus.ts:56]: I tried commenting this out and discovered that we still need this to make auto focus work on the 1st page of the how it works carousel on iOS.

File: src/shared/useAccessibilityAutoFocus.ts [hunk 2]
```diff
@@ -24,34 +24,42 @@ export const focusOnElement = (elementRef: any) => {
...
-    if (!autoFocusRef || !isActive) {
+    if (!isScreenReaderEnabled || !isActive || !isFocus || !isLayoutUpdated || !autoFocusRef) {
       return;
     }
 
-    callable();
-    return cancelable;
-  }, [callable, cancelable, autoFocusRef, isActive]);
+    // Call focus as soon as all considition is met
+    focusOnElement(autoFocusRef);
```
Comment (smcmurtry) [inline src/shared/useAccessibilityAutoFocus.ts:53]: for the onboarding carousel screens this call seems to be what sets the focus

File: src/shared/useAccessibilityAutoFocus.ts (hunk 1 above)
Comment (henrytao-me) [reply src/shared/useAccessibilityAutoFocus.ts:56]: Right. Somehow `AccessibilityInfo.setAccessibilityFocus` is delayed. It needs timeout to actually focus on the element.

**Pull request merged**